contribute/design/1.x/csi/20190107-cStor-volume-resize.md

```diff
@@ -0,0 +1,357 @@
+      {"apiVersion":"v1","kind":"PersistentVolumeClaim","metadata":{"annotations":{},"name":"csi-vol-claim","namespace":"default"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"5Gi"}},"storageClassName":"openebs-csi-sc"}}
+    pv.kubernetes.io/bind-completed: "yes"
+    pv.kubernetes.io/bound-by-controller: "yes"
+    volume.beta.kubernetes.io/storage-provisioner: cstor.csi.openebs.io
+  creationTimestamp: "2019-12-23T13:11:33Z"
+  finalizers:
+  - kubernetes.io/pvc-protection
+  name: csi-vol-claim
+  namespace: default
+  resourceVersion: "7804"
+  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/csi-vol-claim
+  uid: a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: openebs-csi-sc
+  volumeMode: Filesystem
+  volumeName: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+status:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 5Gi
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2019-12-23T13:59:08Z"
+    message: Waiting for filesystem resize.
+    status: "True"
+    type: FileSystemResizePending
+  phase: Bound
+```
+In the above PVC spec capacity is updated to 10Gi but status capacity is 5Gi(Which
+means resize is in progress).
+
+### Implementation Details
+
+The sections below can be assumed to be specific to `cStor` unless mentioned
+otherwise.
+
+### Custom Resources used to resize cStor volume
+
+Sample CVC yaml when resize is in progress
+```yaml
+ apiVersion: openebs.io/v1alpha1
+  kind: CStorVolumeClaim
+  metadata:
+    annotations:
+      openebs.io/volumeID: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+    creationTimestamp: "2019-12-23T13:11:33Z"
+    finalizers:
+    - cvc.openebs.io/finalizer
+    generation: 5
+    labels:
+      openebs.io/cstor-pool-cluster: cstor-sparse-cspc
+    name: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+    namespace: openebs
+    resourceVersion: "7781"
+    selfLink: /apis/openebs.io/v1alpha1/namespaces/openebs/cstorvolumeclaims/pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+    uid: c4e3dfe5-e103-4d8a-8447-a827b710de1c
+  publish:
+    nodeId: 127.0.0.1
+  spec:
+    capacity:
+      storage: 10Gi
+    cstorVolumeRef:
+      apiVersion: openebs.io/v1alpha1
+      kind: CStorVolume
+      name: pvc-a91e400f-ff9b-44ec-b447-bc7bfcfccd80
+      namespace: openebs
+      resourceVersion: "1483"
+      uid: d277113c-8ded-451b-bae6-7c49aa2e346d
+    replicaCount: 1
+  status:
+    capacity:
+      storage: 5Gi
+    condition:
+    - lastTransitionTime: "2019-12-23T13:59:03Z"
+      message: "Resizing cStor volume"
+      reason: ""
+      type: Resizing
+    phase: Bound
+  versionDetails:
+    autoUpgrade: false
+    desired: 1.4.0
+    status:
+      current: 1.4.0
+      dependentsUpgraded: true
+      lastUpdateTime: null
+      state: ""
+```
+
+Sample cstorvolume yaml when resize is in progress
```

Comment:
Similar to above comment. Move this section below the implementation workflow. And create a link to this section, in the note that is mentioning to refer to this.

Reply:
Done